src/components/views/dialogs/MultiInviteDialog.js

```diff
@@ -0,0 +1,199 @@
+/*
+Copyright 2016 OpenMarket Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from 'react';
+
+import {getAddressType, inviteToRoom} from '../../../Invite';
+import sdk from '../../../index';
+
+export default class MultiInviteDialog extends React.Component {
+    constructor(props, context) {
+        super(props, context);
+
+        this._onCancel = this._onCancel.bind(this);
+        this._startInviting = this._startInviting.bind(this);
+
+        this.state = {
+            busy: false,
+            completionStates: [], // State of each address (invited or error)
+            errorTexts: [], // Textual error per address
+            done: false,
+        };
+        for (let i = 0; i < this.props.inputs.length; ++i) {
+            const input = this.props.inputs[i];
+            if (getAddressType(input) === null) {
+                this.state.completionStates[i] = 'error';
+                this.state.errorTexts[i] = 'Unrecognised address';
+            }
+        }
+    }
+
+    _onCancel() {
+        this.props.onFinished(false);
```

Comment:
It would, although this button isn't visible once you start inviting - it probably should be though.